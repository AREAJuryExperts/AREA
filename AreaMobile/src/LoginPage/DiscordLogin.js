import React, { useState, useEffect } from 'react';
import GetImage from '../GetImages/GetImages';
import { Text, Image, StyleSheet, TouchableOpacity } from 'react-native';
import * as WebBrowser from 'expo-web-browser';
import FrontUrl from '../ApiRoute/FrontUrl';
import ApiRoute from '../ApiRoute/ApiRoute';
import * as SecureStore from 'expo-secure-store';

export default function DiscordLogin({}) {
    const [image, setImage] = useState(null);
    const [authUrl, setAuthUrl] = useState(null);

    const getAuthUrl = async () => {
        try {
            let res = await fetch(ApiRoute + '/api/services', {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json'
                },
            })
            if (!res.ok)
                throw new Error("Error while fetching auth url" + await res.text());
            let data = await res.json();
            for (let i in data)
                if (data[i].app === 'Discord' && data[i].loginUrl) {
                    setAuthUrl(data[i].loginUrl);
                    console.log(data[i].loginUrl);
                    return
                }
        } catch (e) {
            console.error(e);
        }
    };

    useEffect(() => {
        setImage(GetImage('discord'))
        getAuthUrl();
    }, []);

    const openService = async () => {
        let jwt = await SecureStore.getItemAsync("AreaToken");
        const params = new URLSearchParams({
            redirect: authUrl,
            jwt : jwt
          });
        let result = await WebBrowser.openBrowserAsync(FrontUrl + "/confirmMobile?" + params.toString());
    };

    return (
        <TouchableOpacity onPress={() => openService()}
        style={{alignItems : 'center', backgroundColor : 'blue', borderRadius : 25, padding : 5, marginTop : 25, width: '100%'}}>
            <Image tintColor={'white'}source={image} style={{ width: 40, height: 40, marginTop: 5, marginBottom : 5 }} />
            <Text style={{color : 'white', fontSize : 20}}>Login with Discord</Text>
        </TouchableOpacity >
    )
};