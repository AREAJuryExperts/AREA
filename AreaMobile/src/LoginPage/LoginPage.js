import { SafeAreaView, StyleSheet, Text, View, StatusBar, TextInput, TouchableOpacity } from 'react-native';
import React, {useState, useEffect} from 'react';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import Logo from '../../assets/logo.svg';
const backColor = "#fff";

export default function LoginPage() {
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [passwordVisible, setPasswordVisible] = useState(false);
  return (
    <View style={styles.container}>
      <StatusBar barStyle="dark-content" backgroundColor={backColor}/>
      <Logo width={150} height={150} />
      <View style={styles.passwordContainer}>
        <TextInput
          style={styles.input}
          onChangeText={text => setUserName(text)}
          value={userName}
          selectionColor={'#0000FF'}
          placeholder='Login' />
          <MaterialCommunityIcons name='account' size={24} color="black" />
      </View>
      <View style={styles.passwordContainer}>
        <TextInput
          style={styles.input}
          onChangeText={text => setPassword(text)}
          value={password}
          selectionColor={'#0000FF'}
          placeholder='Password'
          secureTextEntry={!passwordVisible}/>
        <MaterialCommunityIcons name={passwordVisible ? 'eye-off' : 'eye'} size={24} color="black" onPress={() => setPasswordVisible(!passwordVisible)}/>
      </View>
      <TouchableOpacity onPress={() => console.log("touch !")}>
        <Text style={styles.fgtPassword}>Forgotten password</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.connectionButton}>
        <Text style={styles.connectionButtonText}>Connection</Text> 
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: backColor,
    alignItems: 'center',
    paddingTop: 100,
    paddingBottom: 100,
    paddingLeft: 75,
    paddingRight: 75,
  },
  passwordContainer: { 
    flexDirection: 'row', 
    alignItems: 'center', 
    marginTop: 50, 
},
input : {
  height: 40, 
  borderColor: 'black', 
  borderWidth: 2, 
  width: '100%', 
  borderRadius: 25, 
  paddingLeft: 10, 
  marginRight: 10
},
fgtPassword : {
  marginTop: 50,
  color: 'blue'
},
connectionButton : {
  marginTop: 50,
  backgroundColor: 'blue',
  width: '100%',
  height: 40,
  borderRadius: 25,
  alignItems: 'center',
  justifyContent: 'center'
},
connectionButtonText : {
  color: 'white',
  fontSize : 20
}
});

