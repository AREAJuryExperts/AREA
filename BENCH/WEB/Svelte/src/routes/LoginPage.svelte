<script>
	import { navigate } from 'svelte-routing';
    let email = '';
    let password = '';
  
    const login = () => {
      console.log(`Email: ${email}, Password: ${password}`);
	  navigate('/home')
    };
  
    const forgotPwd = () => {
      navigate('/forgotPassword')
    };
  
  </script>


  <main class="main">
    <img alt="AREA logo" src="./Logo_AREA.png" class="logo">
    <div class="inputContainer">
      <input bind:value={email} class="inputText" placeholder="Email" />
      <input type="password" bind:value={password} class="inputText" placeholder="Password" />
    </div>
    <div class="forgotPwd">
      <!-- <a class="forgotPwdButton" on:click={forgotPwd}>Forgot password ?</a> -->
	  <button on:click={forgotPwd} class="forgotPwdButton">
		<span>Forgot password ?</span>
	</button>
    </div>
    <button on:click={login} class="buttonLogin">Login</button>
      </main>




  <style>
    .main {
      display: flex;
      justify-content: center;
      align-items: center;
      min-height: 100vh;
      flex-direction: column;
    }
  
    .inputContainer {
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;
      margin-top: 20px;
      margin-bottom: 10px;
    }
  
    .inputText {
      width: 400px;
      border-radius: 5px;
      border: 1px solid #363636;
      margin: 10px;
      padding: 10px;
    }
  
    .logo {
      height: 150px;
      pointer-events: none;
    }
  
    .buttonLogin {
      width: 200px;
      border: none;
      border-radius: 5px;
      margin: 10px;
      padding: 10px;
      background-color: #0000FF;
      color: white;
      font-weight: bold;
      cursor: pointer;
      box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.2);
    }
  
    .forgotPwd {
      width: 400px;
      display: flex;
      justify-content: flex-start;
      align-items: center;
    }

	.forgotPwdButton {
    color: #0000FF;
    text-decoration: none;
    padding: 10px;
    cursor: pointer;
	border: none;
	background: none;
  }
  .forgotPwdButton:hover {
    text-decoration: underline;
  }
  
  </style>