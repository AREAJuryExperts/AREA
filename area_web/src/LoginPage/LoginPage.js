import style from "./LoginPage.module.css";
import React, { useState } from 'react';
import { useNavigate } from "react-router-dom";

import LogoAREA from "../assets/Logo_AREA.png";
import TextField from '@mui/material/TextField';
import { API_URL } from "../utils";

function TextsFields({ email, setEmail, password, setPassword }) {
    return (
        <div className={style.textFieldsContainer}>
            <div className={style.textFieldContainer}>
                <TextField
                    label="Email"
                    variant="outlined"
                    type="email"
                    className={style.inputTextField}
                    value={email}
                    onChange={(e) => setEmail(e.target.value)}
                />
            </div>
            <div className={style.textFieldContainer}>
                <TextField
                    label="Password"
                    variant="outlined"
                    type="password"
                    className={style.inputTextField}
                    value={password}
                    onChange={(e) => setPassword(e.target.value)}
                />
            </div>
        </div>
    );
}

export default function Login() {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [errorMessage, setErrorMessage] = useState("");
    

    const navigate = useNavigate();

    const handleLogin = () => {
        let params = (new URL(document.location)).searchParams;
        let redirect_uri = params.get("redirect");

        fetch(API_URL + "/auth/login", {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Access-Control-Allow-Origin": "*",
            },
            body: JSON.stringify({
                email: email,
                password: password,
            }),
        })
            .then((response) => response.json())
            .then((data) => {
                if (data.msg === "ok") {
                    localStorage.setItem("jwt", data.jwt); 
                    if (redirect_uri)
                        window.location.href = redirect_uri;
                    else
                        window.location.href = "/"
                } else if (data.msg === "User not confirmed") {
                    setErrorMessage("Veuillez confirmer votre compte");
                } else if (data.msg === "Invalid credentials") {
                    setErrorMessage("Utilisateur inconnu ou mot de passe incorrect");
                } else {
                    setErrorMessage("Erreur lors de la connexion au serveur");
                }
            })
            .catch((error) => {
                console.error("Error:", error);
            });
    };

    const handleNotRegistered = () => {
        navigate("/register");
    }

    const handleForgottenPassword = () => {
        console.log("pwd forgotten");
    }

    return (
        <div className={style.MainContainerLoginPage}>
            <img src={LogoAREA} alt="Logo AREA" className={style.logoLoginPage} />
            <TextsFields email={email} setEmail={setEmail} password={password} setPassword={setPassword} />
            <span style={{color:'red'}}>{errorMessage}</span>
            <div className={style.forgottenPasswordButtonContainer}>
                <button onClick={handleForgottenPassword} className={style.forgottenPasswordButton}>Forgot password ?</button>
            </div>
            <button className={style.loginButton} onClick={handleLogin}>Login</button>
            <div className={style.notRegisteredButtonContainer}>
                <button onClick={handleNotRegistered} className={style.notRegisteredButton}>Not registered for the moment</button>
            </div>
            <button className={style.discordButton} onClick={() => {window.location.href = "https://discord.com/api/oauth2/authorize?client_id=1188091323908112475&response_type=code&redirect_uri=https%3A%2F%2Farea.david-benistant.com%2FconfirmDiscordLogin&scope=identify+email"}} >Login with Discord</button>
        </div>
    );
}
