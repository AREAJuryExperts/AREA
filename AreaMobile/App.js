import { useEffect, useState } from 'react';
import LoginPage from './src/LoginPage/LoginPage';
import HomePage from './src/HomePage/HomePage';
import RegisterPage from './src/RegisterPage/RegisterPage';
import * as SecureStore from 'expo-secure-store';
import * as Linking from 'expo-linking';

// const prefix = Linking.createURL('/');
export default function App() {
  // const linking = {
  //   prefixes: [prefix],
  // };
  const [currentScreen, setCurrentScreen] = useState('login');
  const [registerInfo, setRegisterInfo] = useState('');
  const [deepLinkReceived, setDeepLinkReceived] = useState(false);
  useEffect(() => {
    SecureStore.getItemAsync("AreaToken").then((token) => {
      if (token) {
        setCurrentScreen('home');
      }
      else
        setCurrentScreen('login');
    })
  }, []);
  useEffect(() => {
    const handleDeepLink = event => {
      console.log('event', event);
        setDeepLinkReceived(!deepLinkReceived);
        let url = event.url;
        if (typeof url === 'undefined')
          return;
        if (url.includes("jwt")) {
          let jwt = url.split("jwt=")[1];
          console.log('jwt', jwt);
          SecureStore.setItemAsync("AreaToken", jwt).then(() => {
            setCurrentScreen('home');
          })
        }
    };
    Linking.getInitialURL().then((url) => {
      if (url) handleDeepLink(url);
    });
    // Add event listener for incoming links
    Linking.addEventListener('url', handleDeepLink);
}, []);
  return (
    <>
      {
        (currentScreen === 'login' && <LoginPage setCurrentScreen={setCurrentScreen} registerInfo={registerInfo} setRegisterInfo={setRegisterInfo}/>) ||
        (currentScreen === 'register' && <RegisterPage setCurrentScreen={setCurrentScreen} setRegisterInfo={setRegisterInfo}/>) ||
        (currentScreen === 'home' && <HomePage setCurrentScreen={setCurrentScreen} deepLinkReceived={deepLinkReceived}/>)
      }
    </>
  )
}

