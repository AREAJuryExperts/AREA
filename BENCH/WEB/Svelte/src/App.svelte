<script>
  export let name;
  import { onMount } from 'svelte';
  import { navigate, Router, Route} from 'svelte-routing';
  import LoginPage from './routes/LoginPage.svelte';
  import HomePage from './routes/HomePage.svelte';
  import ForgotPassword from './routes/ForgotPassword.svelte'

  onMount(() => {
    navigate('/');
  });
</script>

<Router>
  <Route path="/" component={LoginPage} />
  <Route path="/home" component={HomePage}/>
  <Route path="/forgotPassword" component={ForgotPassword}/>
</Router>