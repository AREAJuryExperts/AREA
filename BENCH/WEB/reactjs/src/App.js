import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import React from "react";
import LoginPage from "./Login";
import HomePage from "./Home";
import ForgotPwdPage from "./ForgotPwd";

function App() {

    return (
        <Router>
            <Routes>
                <Route path="/" element={<LoginPage />} />
                <Route path="/home" element={<HomePage />} />
                <Route path="/forgotPwd" element={<ForgotPwdPage />} />
            </Routes>
        </Router>
    );
}

export default App;