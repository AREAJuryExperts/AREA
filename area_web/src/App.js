import React from "react";

//Import for navigation through pages
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";

//Pages
import LoginPage from "./LoginPage/LoginPage";
import HomePage from "./HomePage/HomePage";
import RegisterPage from "./RegisterPage/RegisterPage";
import Confirm from "./Confirm/confirm";
import WaitingPage from "./RegisterPage/WaitingPage";
import ConfirmTrello from "./Confirm/confirmTrello";
import ConfirmDiscord from "./Confirm/confirmDiscord";

function App() {

    return (
        <Router>
            <Routes>
                <Route path="/" element={<LoginPage />} />
                <Route path="/register" element={<RegisterPage />} />
                <Route path="/home" element={<HomePage />} />
                <Route path="/confirm" element={<Confirm />} />
                <Route path='/confirmTrello' element={<ConfirmTrello />} />
                <Route path="/confirmDiscord" element={<ConfirmDiscord />} />
                <Route path="/waitingConfirmation" element={<WaitingPage />} />
            </Routes>
        </Router>
    );
}

export default App;
