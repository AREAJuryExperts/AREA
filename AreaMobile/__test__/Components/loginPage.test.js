import React from 'react';
import renderer from 'react-test-renderer';
import LoginPage from '../../src/LoginPage/LoginPage.js';
import App from '../../App.js';
import {fireEvent, cleanup, render, screen, act} from '@testing-library/react-native';

afterEach(cleanup);

let registerInfo = '';
const setRegisterInfo = (info) => {
}

describe('<Login />', () => {
  it('renders correctly', async () => {
    const tree = renderer.create(<LoginPage registerInfo={registerInfo} setRegisterInfo={setRegisterInfo}/>).toJSON();
    expect(tree).toMatchSnapshot();
    return
  });
  it('can fill prompts and click buttons', async () => {
    const {getByTestId} = render(<LoginPage registerInfo={registerInfo} setRegisterInfo={setRegisterInfo}/>);
    const {getByText} = screen;
    const foundButton = getByTestId("loginBtn");
    const foundPasswordInput = getByTestId("passwordInput");
    const userNameInput = getByTestId("userNameInput");
    fireEvent.changeText(userNameInput, 'robin.denni@epitech.eu');
    fireEvent.changeText(foundPasswordInput, 'C00lPassword');
    fireEvent.press(foundButton);
    expect(getByText('Login')).toBeTruthy();
    return
  });
  it('can go to the register page', async () => {
    const {getByTestId} = render(<App />);
    const {getByText} = screen;
    act(() => {
      const foundButton = getByTestId("registerBtn");
      fireEvent.press(foundButton);
    })
    expect(getByText('Password confirmation')).toBeTruthy();
    return;
  })
});