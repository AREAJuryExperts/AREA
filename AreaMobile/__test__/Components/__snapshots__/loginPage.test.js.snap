// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "height": 3001.5,
      "paddingBottom": 100,
      "paddingLeft": 75,
      "paddingRight": 75,
      "paddingTop": 35,
      "width": "100%",
    }
  }
  scrollEnabled={false}
>
  <View>
    <Mocked SVG
      height={150}
      width={150}
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 50,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
        selectionColor="#0000FF"
        style={
          {
            "borderColor": "black",
            "borderRadius": 15,
            "borderWidth": 2,
            "color": "black",
            "height": 40,
            "marginRight": 10,
            "paddingLeft": 10,
            "width": "100%",
          }
        }
        testID="userNameInput"
        value=""
      />
      <Text />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 50,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Mot de passe"
        secureTextEntry={true}
        selectionColor="#0000FF"
        style={
          {
            "borderColor": "black",
            "borderRadius": 15,
            "borderWidth": 2,
            "color": "black",
            "height": 40,
            "marginRight": 10,
            "paddingLeft": 10,
            "width": "100%",
          }
        }
        testID="passwordInput"
        value=""
      />
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="registerBtn"
    >
      <Text
        style={
          {
            "color": "blue",
            "marginTop": 20,
          }
        }
      >
        Not registered yet?
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "blue",
          "borderRadius": 25,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 50,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="loginBtn"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "blue",
          "borderRadius": 25,
          "marginTop": 25,
          "opacity": 1,
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <Image
        source={null}
        style={
          {
            "height": 40,
            "marginBottom": 5,
            "marginTop": 5,
            "width": 40,
          }
        }
        tintColor="white"
      />
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Login with Discord
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
